Sources/GRPCCore/Call/Client/ClientInterceptorPipelineOperation.swift

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A `ClientInterceptorPipelineOperation` describes to which RPCs a client interceptor should be applied.
+///
+/// You can configure a client interceptor to be applied to:
+/// - all RPCs and services;
+/// - requests directed only to specific services; or
+/// - requests directed only to specific methods (of a specific service).
+///
+/// - SeeAlso: ``ClientInterceptor`` for more information on client interceptors, and
+///  ``ServerInterceptorPipelineOperation`` for the server-side version of this type.
+public struct ClientInterceptorPipelineOperation: Sendable {
```

Comment:
Not sure how I missed this for the server, but we should add tests for this specific type as well - `applies(to:)` is the only thing we need to test here though.